apps/bilateral_grid/Makefile

```diff
@@ -1,22 +1,22 @@
 include ../support/Makefile.inc
 
-all: filter
+all: $(BIN)/filter
 
-bilateral_grid: bilateral_grid.cpp
-	$(CXX) $(CXXFLAGS) bilateral_grid.cpp -g $(LIB_HALIDE) -o bilateral_grid -lz $(LDFLAGS) \
-	$(LLVM_SHARED_LIBS)
+$(BIN)/bilateral_grid.a: bilateral_grid_generator.cpp
+	@-mkdir -p $(BIN)
+	$(HALIDE_SRC_PATH)/tools/gengen.sh -c $(CXX) -l $(LIB_HALIDE) -o $(BIN) -s $^ target=host
```

Comment:
I'm hoping these Makefiles can teach people how generators work (or at least reinforce the tutorial), so it might be most educational to expand it into two rules - one that compiles the generator, and one that runs it.